GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1687,18 +1691,11 @@ private void RevisionsCellPainting(object sender, DataGridViewCellPaintingEventA
                         int gravatarTop = e.CellBounds.Top + textHeight + 6;
                         int gravatarLeft = e.CellBounds.Left + baseOffset + 2;
 
-
-                        Image gravatar = Gravatar.GravatarService.GetImageFromCache(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, FallBackService.MonsterId);
-
-                        if (gravatar == null && !string.IsNullOrEmpty(revision.AuthorEmail))
-                        {
-                            ThreadPool.QueueUserWorkItem(o =>
-                            Gravatar.GravatarService.LoadCachedImage(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, null, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, RefreshGravatar, FallBackService.MonsterId));
-                        }
-
+                        var gravatar = await _gravatarService.GetAvatarAsync(revision.AuthorEmail, gravatarSize, DefaultImageType.MonsterId);
                         if (gravatar != null)
+                        {
                             e.Graphics.DrawImage(gravatar, gravatarLeft + 1, gravatarTop + 1, gravatarSize, gravatarSize);
```

Comment:
Indeed it is happening. I think you may be right - the graphics object could probably be disposed by the time the image is loaded...
The cardview has never worked for me - it would just start chewing the cpu... However today, when I tried to repro this problem, the cardview worked. Go figure...